tfjs-backend-wasm/src/cc/sin_cos_workaround.cc

```diff
@@ -11,35 +11,67 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  * ===========================================================================*/
-#include <math.h>
+
+#include <cmath>
 
 #include "tfjs-backend-wasm/src/cc/sin_cos_workaround.h"
 
 namespace tfjs {
 namespace sin_cos_workaround {
 
-float sin_fixed(float x) {
-  if (isnan(x)) return nan("");
-  auto zero_to_2pi = fmod(fmod(x, 2 * M_PI) + 2 * M_PI, 2 * M_PI);
-
-  if (zero_to_2pi < M_PI_4) {
-    return sin(zero_to_2pi);
-  } else if (zero_to_2pi < M_PI_2) {
-    auto past_pi_4 = zero_to_2pi - M_PI_4;
-    return cos(M_PI_4 - past_pi_4);
-  } else if (zero_to_2pi < M_PI) {
-    auto past_pi_2 = zero_to_2pi - M_PI_2;
-    return sin_fixed(M_PI_2 - past_pi_2);
+namespace {
+
+template <typename T>
+inline T ShiftRadianToZeroTo2PI(const T& x) {
+  if (std::isnan(x)) {
+    return x;
+  }
+  return std::fmod(std::fmod(x, 2 * M_PI) + 2 * M_PI, 2 * M_PI);
+}
+
+template <typename T>
+inline T SinZeroTo2PI(T x) {
+  if (std::isnan(x)) {
+    return x;
+  }
+
+  if (x < M_PI_4) {
+    return std::sin(x);
+  } else if (x < M_PI_2) {
+    return std::cos(M_PI_2 - x);
+  } else if (x < M_PI) {
+    return SinFixedImpl<T, /*is_shifted=*/true>(M_PI - x);
   } else {
-    return -sin_fixed(2 * M_PI - zero_to_2pi);
+    return -SinFixedImpl<T, /*is_shifted=*/true>(2 * M_PI - x);
```

Comment:
[SinFixedImpl is missing](https://source.cloud.google.com/results/invocations/80cf68bf-f174-42b0-a8d6-d7c31381f6ad/targets/%2F%2Ftfjs-backend-wasm:tfjs-backend-wasm_pkg/log). Is there a missing file in this PR?